### PR TITLE
[CHK-597] feat(retry-queue): Add retry queue for messages that fail to be delivered

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

* Added retry queue for messages that incur in errors when sending through AWS SES
* Refactored `sendMail` function in functional style

#### Motivation and Context

Refer to updated design considerations on error handling.

**NOTE:** This PR purposefully does not include a backup error queue, it will be added in a later PR based upoon feedback on this one.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/controllers/EmailsControllers.ts [hunk 1]
```diff
@@ -209,34 +270,75 @@ export function sendMail(
...
+          mailTrasporter,
+          config,
+          retryCount - 1
+        );
+      }
+    }
   };
-}
+
+  setInterval(retrieveMessage, 1000);
```
Comment (giovanniberti) [inline src/controllers/EmailsControllers.ts:343]: Should also queue polling time be a configuration variable?

File: api-spec/index.yaml [hunk 2]
```diff
@@ -37,6 +37,11 @@ paths:
           examples:
             application/json:
               importoSingoloVersamento: 200
+        "202":
+          description: Email notification enqueued
+          examples:
```
Comment (infantesimone) [inline api-spec/index.yaml:42]: to delate example

File: src/controllers/EmailsControllers.ts [hunk 3]
```diff
@@ -209,34 +273,77 @@ export function sendMail(
...
+    );
+}
 
-    const maybeClientId = getClientId(req);
-    if (E.isLeft(maybeClientId)) {
-      const error = maybeClientId.left;
-      return headerValidationErrorHandler(error);
-    }
-    const clientId = maybeClientId.right as NotificationsServiceClientEnum;
+export const addRetryQueueListener = (
```
Comment (infantesimone) [inline src/controllers/EmailsControllers.ts:310]: why define it in the controller?

**Review by infantesimone — APPROVED** [review]
just few comments, but it sounds good

File: src/controllers/EmailsControllers.ts (hunk 3 above)
Comment (infantesimone) [reply src/controllers/EmailsControllers.ts:310]: maybe in a dedicated file, what do you think?

Comment (giovanniberti) [reply src/controllers/EmailsControllers.ts:310]: I'm ok with that, the question was more about the naming 😄 (and the parent directory)

Comment (infantesimone) [reply src/controllers/EmailsControllers.ts:310]: `src/queues/RetryListener.ts` or something like that?

**Review by infantesimone — APPROVED** [review]

**Pull request merged**